src/net/http/httputil/reverseproxy.go

```diff
@@ -244,6 +259,17 @@ func (p *ReverseProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 		outreq.Header.Set("Upgrade", reqUpType)
 	}
 
+	if p.OverwriteForwardedHeaders {
+		proto := "https"
+		if req.TLS == nil {
```

Comment:
Not sure about this. `req.TLS` is only set when the current http server is handling the TLS. But if TLS is done by a load balancer in front of this server, it will say `http`, how can the backend know if it was an `https` request originally?

Reply:
Then you'll have to handle this manually. If the TLS termination is handled by a non-HTTP server, the proxy cannot guess it.